scio-core/src/main/scala/com/spotify/scio/coders/instances/ScalaCoders.scala

```diff
@@ -335,6 +336,75 @@ private[coders] class MapCoder[K, V](val kc: BCoder[K], val vc: BCoder[V])
   override def getCoderArguments: JList[_ <: BCoder[_]] = List(kc, vc).asJava
 }
 
+private[coders] class SortedMapCoder[K: Ordering, V](
+  val kc: BCoder[K],
+  val vc: BCoder[V]
+) extends StructuredCoder[SortedMap[K, V]] {
+  private[this] val lc = VarIntCoder.of()
+
+  override def encode(value: SortedMap[K, V], os: OutputStream): Unit = {
+    lc.encode(value.size, os)
+    val it = value.iterator
+    while (it.hasNext) {
+      val (k, v) = it.next()
+      kc.encode(k, os)
+      vc.encode(v, os)
+    }
+  }
+
+  override def decode(is: InputStream): SortedMap[K, V] = {
+    val l = lc.decode(is)
+    val builder = SortedMap.newBuilder[K, V]
+    builder.sizeHint(l)
+    var i = 0
+    while (i < l) {
+      val k = kc.decode(is)
+      val v = vc.decode(is)
+      builder += (k -> v)
+      i = i + 1
+    }
+    builder.result()
+  }
+
+  // delegate methods for determinism and equality checks
+  override def verifyDeterministic(): Unit =
+    throw new NonDeterministicException(
+      this,
+      "Ordering of entries in a SortedMap may be non-deterministic."
```

Comment:
This is probably still true if two elements are equal according to `Ordering[K]`?

Reply:
Map can't have 2 keys equal (`get` must return a single value). This can be set as `verifyDeterministic`

Reply:
ah cool yeah, I tried this:
```
scala> SortedMap((5, 2), (3, 8))((_, _) => 0)
val res0: scala.collection.immutable.SortedMap[Int,Int] = TreeMap(5 -> 8)
```